hat/app/org/hatdex/hat/api/service/applications/ApplicationsService.scala

```diff
@@ -158,17 +161,18 @@ class ApplicationsService @Inject() (
     maybeDataDebitSetup.getOrElse(Future.successful(Done)) // If data debit was there, must have been setup successfully
   }
 
-  private def joinContract(application: Application, hatName: String): Future[Any] =
+  def joinContract(application: Application, hatName: String): Future[Any] = {
     application.kind match {
-      case ApplicationKind.Contract(x) =>
+      case _: ApplicationKind.Contract =>
         adjudicatorClient
           .joinContract(
             hatName,
-            io.dataswift.adjudicator.Types
-              .ContractId(
-                java.util.UUID.fromString(application.id)))
-      case _ => Future.successful(Unit)
+            ContractId(UUID.fromString(application.id)))
+      case _ => {
+        Future.successful(Unit)
```

Comment:
Suggest replacing `Unit` with `akka.Done`. Successful Future that returns a Unit can lead to unexpected results, especially when flattening Futures.